File: pset6_Python/mario_more.py
from cs50 import get_int

# Obtém e valida se a altura de blocos informada pelo usuário é válida
while True:
    height = get_int("Digite a altura entre 1 e 8: ")
    if height < 1 or height > 8:
        height = get_int("Digite a altura entre 1 e 8: ")
    else:
        break

# Imprime os blocos com base na altura
for i in range(height):
    print((height - 1 - i) * " ", end="")
    print((i + 1) * "#", end="")
    # Segunda pirâmide
    print("  ", end="")
    print((i + 1) * "#")

File: pset1_c/mario_more.c
#include <stdio.h>
#include <cs50.h>

int main (void)
{
    // Obtém a quantidade de linhas de blocos (#) que serão impressos.
    int altura;
    do 
    {
        altura = get_int ("Digite um número entre 1 e 8: ");  
    }
    while (altura < 1 || altura > 8);
        
    // Laço de repetição para impressão das pirâmides
    for (int linha = 0; linha < altura; linha++)
    {
        // Impressão da primeira pirâmide
        for (int coluna = 0; coluna < altura; coluna++)
        {
            // Verificação dos espaços para alinhamento dos blocos
            if (linha + coluna < altura - 1)
            {
              printf (" ");  
            }
            else 
            {
              printf ("#");   
            }
        }
        printf ("  ");
        
        // Impressão da segunda pirâmide
        for (int coluna = 0; coluna < altura; coluna++)
        {
            if (linha >= coluna)
            {
              printf ("#");
            }
        }
        printf ("\n");
    }
}
